gnat2goto/driver/tree_walk.adb

```diff
@@ -4836,40 +4836,52 @@ package body Tree_Walk is
    function Make_Array_Index_Op
      (Base_Irep : Irep; Base_Type : Node_Id; Idx_Irep : Irep) return Irep
    is
+      Source_Loc : constant Source_Ptr := Sloc (Base_Type);
       First_Irep : constant Irep :=
         Make_Array_First_Expr (Base_Type, Base_Irep);
-      Zero_Based_Index : constant Irep := New_Irep (I_Op_Sub);
+      Zero_Based_Index : constant Irep :=
+        Make_Op_Sub (Rhs             => First_Irep,
+                     Lhs             => Idx_Irep,
+                     Source_Location => Source_Loc,
+                     Overflow_Check  => False,
+                     I_Type          => Get_Type (Idx_Irep),
+                     Range_Check     => False);
       Result_Type : Irep;
-      Data : constant Irep := New_Irep (I_Member_Expr);
-      Offset : constant Irep := New_Irep (I_Op_Add);
-      Deref : constant Irep := New_Irep (I_Dereference_Expr);
       Pointer_Type : constant Irep := New_Irep (I_Pointer_Type);
+      Data : constant Irep :=
+        Make_Member_Expr (Compound         => Base_Irep,
+                          Source_Location  => Source_Loc,
+                          Component_Number => 3,
```

Comment:
magic numbers need comments :)
e.g. a few lines previously:
"Accessing element 3 of the data structure that represents/wraps arrays." Element 1 and 2 are the lower and upper bounds, element 3 is a pointer to the array itself"

Reply:
@xbauch as you've pointed out there's many locations with this will occur. Your call.

Reply:
I attached the comment to the first occurrence, i.e. in `Build_Array_Lit_Func_Body`.